### DISCUSSION TITLE
Specifying Hall Number in spglib

### DISCUSSION THREAD

**Discussion by Qihan777** [spglib/spglib#480] [Q&A] [answered]
In spglib.get_symmetry_dataset() function, without specifying this option, always the first one (the smallest serial number corresponding to the space-group-type in [list of space groups (Seto’s web site)](https://yseto.net/?page_id=29%3E)) among possible choices and settings is chosen as default. How to specify a particular Hall number in the spglib.get_symmetry_dataset() function? For instance, if a material corresponds to the second Hall number rather than the first (e.g., Hall numbers 436 and 437), how can I input this specific Hall number to access the correct Wyckoff positions?

**Comment by lan496** [comment]
Can you try to specify `hall_number` as follows?
```python
dataset = spglib.get_symmetry_dataset(cell, hall_number=437)
```
https://spglib.readthedocs.io/en/stable/api/python-api/spglib/spglib.html#spglib.get_symmetry_dataset

**Reply by Qihan777** [reply]
Thank for helping. Yes, I used it to access the symmetry dataset with specify hall number. However, I need to determine the true Hall number for a given material structure for my future studies. The limitation of `spglib.get_symmetry_dataset()` function seems to be that it only return the first Hall number (the smallest serial number corresponding to the space group type). How can I obtain the correct Hall number if I don't initially know it?

**Reply by lan496** [reply, marked as answer]
I am not sure of the meaning of "the correct Hall number" you said, but you may need to tabulate a table from Hall number (1-530) to ITA's space-group type number (1-230). For example, pyxtal seems to have such a mapping.
https://github.com/qzhu2017/PyXtal/blob/1ba044cace1815d450e476a1fcb2fe8cb5798923/doc/Settings.rst#space-group
